mk/docs.mk

```diff
@@ -288,3 +288,13 @@ doc/style/index.html: $(RUSTBOOK_EXE) $(wildcard $(S)/src/doc/style/*.md) | doc/
 	@$(call E, rustbook: $@)
 	$(Q)rm -rf doc/style
 	$(Q)$(RUSTBOOK) build $(S)src/doc/style doc/style
+
+ERR_IDX_GEN = $(CFG_BUILD)/error-index-generator
+RUSTC_EXE = $(HBIN2_H_$(CFG_BUILD))/rustc
+error-index: doc/error-index.html
+
+doc/error-index.html: src/etc/error-index-generator.rs $(RUSTC_EXE) | doc/
+	@$(call E, error-index-generator: $@)
+	$(Q)LD_LIBRARY_PATH=$(HLIB2_H_$(CFG_BUILD)) $(RUSTC_EXE) $< -o $(ERR_IDX_GEN)
```

Comment:
Unfortunately `LD_LIBRARY_PATH` only works on linux, but you should be able to replace this with `RPATH_VAR2_T_$(CFG_BUILD)_H_$(CFG_BUILD)` (see `RUSTBOOK` at the top of this file for more information).

Also, this should be split into two separate steps, one for building `error-index-generator` and one for running it.